### ISSUE TITLE
 Cannot find module 'zod/lib/src/parser' 

### ISSUE THREAD

**Issue by desfero** [react-hook-form/resolvers#220]
**Describe the bug**
The latest resolvers package with `zod` gives a typescript error.

```
node_modules/@hookform/resolvers/zod/dist/types.d.ts:3:34 - error TS2307: Cannot find module 'zod/lib/src/parser' or its corresponding type declarations.

3 import type { ParseParams } from 'zod/lib/src/parser';
```

**To Reproduce**
1. Install latest version of `@hookform/resolvers` and `zod` in a TS project
2. Run `tsc` to get the error. (not `skipLibCheck` should be `false`)

**Expected behavior**
No ts error.

**Additional context**
`ParseParams` can be now imported directly from `zod` root to avoid any potential directory structure changes in the future

**Comment by jorisre** [comment]
Thank you for reporting, someone already reported it: https://github.com/react-hook-form/resolvers/issues/154

We currently support Zod v1, here is the PR for Zod v3: #200

**Comment by desfero** [comment]
oh, I thought that one from April was already resolved.
Do you need any help with the migration? 
I found also that it's not very clear that Zod v3 is not yet supported.

**Comment by jorisre** [comment]
Fixed with 2.8.0!
